deployment/cdk/opensearch-service-migration/lib/migration-services-yaml.ts

```diff
@@ -10,16 +10,44 @@ export class MetricsSourceYaml {
     cloudwatch? : object | null = null;
 }
 
+export class ECSService {
+    cluster_name: string | undefined;
+    service_name: string | undefined;
+    aws_region: string | undefined;
+    constructor() {
+        this.cluster_name = undefined;
+        this.service_name = undefined;
+        this.aws_region = undefined;
+    }
+}
+
+export class RFSBackfillYaml {
+    ecs: ECSService;
+    constructor() {
+        this.ecs = new ECSService();
+    }
+}
+
+export class OSIBackfillYaml {
+
+}
+
 export class ServicesYaml {
     source_cluster: ClusterYaml;
     target_cluster: ClusterYaml;
     metrics_source: MetricsSourceYaml = new MetricsSourceYaml();
+    backfill: RFSBackfillYaml | OSIBackfillYaml;
 
     stringify(): string {
         return yaml.stringify({
             source_cluster: this.source_cluster,
             target_cluster: this.target_cluster,
-            metrics_source: this.metrics_source
+            metrics_source: this.metrics_source,
+            backfill: this.backfill
+            ? (this.backfill instanceof RFSBackfillYaml
```

Comment:
Its strange that this logic is in `ServicesYaml`.  By using polymorphism the different types can self-describe how they are formatted as json internally